models/forum.js

```diff
@@ -1,17 +1,34 @@
 const mongoose = require('mongoose');
 
 const forumSchema = new mongoose.Schema({
+  parentId: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Forum',
+    default: null,
+    validate: {
+      validator: function(id) {
+        if (this.isAnswer && !id) {
+          return false;
+        } else {
+          return true;
+        }
+      },
+      message: 'parentId is reuired for an answers'
+    }
+  },
   title: {
```

Comment:
Also validate whether title is given if it is a question (`! this.isAnswer && !! title`).

Reply:
The title is compulsory for both Q and A as of right now

Reply:
There should be no title for answer, but there should be a title for a question, similar to stack overflow/ quora.